extensions/wikia/Discussions/maintenance/SiteStatus.class.php

```diff
@@ -0,0 +1,411 @@
+	private $discussionsEnabled;
+	private $navigationEnabled;
+	private $forumsEnabled;
+	private $numThreadedForumPosts;
+	private $numWikiForumPosts;
+
+	/** @var \DateTime */
+	private $discussionsFound;
+	/** @var \DateTime */
+	private $firstPostFound;
+	/** @var \DateTime */
+	private $lastMigrated;
+	/** @var \DateTime */
+	private $lastUpdated;
+
+	public static function run( \DatabaseBase $db, $test = false, $verbose = false, $params = [] ) {
+		$status = new SiteStatus(
+			$db,
+			$verbose,
+			$test,
+			$params[\RunOnCluster::PARAM_SITE_ID],
+			$params[\RunOnCluster::PARAM_DB_NAME]
+		);
+
+		// Don't do anything if this has been recently updated by this script
+		if ( $status->skipCataloging() ) {
+			return;
+		}
+
+		$status->update();
+		$status->writeBack();
+	}
+
+	public function __construct( $localDbh, $verbose, $test, $siteId, $dbName ) {
+		$this->localDbh = $localDbh;
+		$this->verbose = $verbose;
+		$this->test = $test;
+		$this->siteId = $siteId;
+		$this->dbName = $dbName;
+
+		$this->debug("== Checking status for site $siteId / $dbName");
+
+		$this->load();
+	}
+
+	public function debug( $msg ) {
+		if ( empty( $this->verbose ) ) {
+			return;
+		}
+
+		echo $msg."\n";
+	}
+
+	private function load() {
+		( new \WikiaSQL() )
+			->SELECT( "*" )
+			->FROM( self::TABLE_STATUS )
+			->WHERE( 'site_id' )->EQUAL_TO( $this->siteId )
+			->run(
+				$this->getCentralDbr(),
+				function( $result ) {
+					/** @var \ResultWrapper|bool $result */
+					if ( !is_object( $result ) ) {
+						return;
+					}
+
+					$row = $result->fetchObject();
+
+					if ( empty( $row ) ) {
+						return;
+					}
+
+					$this->hasStatusRecord = true;
+					$this->status = $row->status;
+					$this->siteAvailable = $row->site_available;
+					$this->discussionsEnabled = $row->discussions_enabled;
+					$this->navigationEnabled = $row->navigation_enabled;
+					$this->forumsEnabled = $row->forums_enabled;
+					$this->numThreadedForumPosts = $row->tf_posts;
+					$this->numWikiForumPosts = $row->wf_posts;
+					$this->discussionsFound = new \DateTime( $row->discussions_found );
+					$this->firstPostFound = new \DateTime( $row->first_post_found );
+					$this->lastMigrated = new \DateTime( $row->last_migrated );
+					$this->lastUpdated = new \DateTime( $row->last_updated );
+				}
+			);
+
+		$this->cityInfo = \WikiFactory::getWikiByDB( $this->dbName );
+	}
+
+	public function update() {
+		$this->probeSiteAvailable();
+		$this->updateWikiVariables();
+		$this->findExistingPosts();
+		$this->probeDiscussions();
+	}
+
+	public function writeBack() {
+		$this->debug( "Writing content to " . self::TABLE_STATUS );
+
+		$statement = new \WikiaSQL();
+		if ( $this->hasStatusRecord ) {
+			$statement
+				->UPDATE( self::TABLE_STATUS )
+				->WHERE( 'site_id' )->EQUAL_TO( $this->siteId );
+		} else {
+			$statement
+				->INSERT( self::TABLE_STATUS )
+				->SET( 'site_id', $this->siteId );
+		}
+
+		$now = (new \DateTime())->format('Y-m-d H:i:s');
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Y-m-d H:i:s" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)